static/js/lightbox.js

```diff
@@ -210,14 +210,23 @@ function display_image(payload) {
     img.src = payload.source;
     $img_container.html(img).show();
 
-    const filename = payload.url?.split("/").pop();
+    const url = new URL(payload.url, window.location.href);
+    const filename = url.pathname.split("/").pop();
     $(".image-description .title")
         .text(payload.title || "N/A")
         .attr("aria-label", payload.title || "N/A")
         .prop("data-filename", filename || "N/A");
     $(".image-description .user").text(payload.user).prop("title", payload.user);
 
-    $(".image-actions .open, .image-actions .download").attr("href", payload.source);
+    $(".image-actions .open").attr("href", payload.source);
+    const same_origin = url.origin == window.location.origin;
+    if (same_origin && url.pathname.startsWith("/user_uploads/")) {
+        // Switch to the "download" handler, so S3 URLs set their Content-Disposition
+        url.pathname = url.pathname.replace("/user_uploads/", "/user_uploads/download/");
+        $(".image-actions .download").attr("href", url.href);
+    } else {
+        $(".image-actions .download").attr("href", payload.source);
```

Comment:
Perhaps hide the Download button here; it’s just going to be a duplicate of the Open button.

Reply:
That'll hide the download even when, say, it's a camo link on the same host, which would download just fine.

I'll hide it in a `else if (!same_origin) { ... }`